Projects/UOContent/Network/Packets/IncomingPlayerPackets.cs

```diff
@@ -420,27 +422,40 @@ public static void DisplayGumpResponse(NetState state, SpanReader reader)
                 return;
             }
 
-            var textEntries = new TextRelay[textCount];
+            scoped Span<ushort> textIds = default;
+            scoped Span<Range> textRanges = default;
+            ReadOnlySpan<byte> rawTextData = default;
 
-            for (var i = 0; i < textEntries.Length; ++i)
+            if (textCount > 0)
             {
-                int entryID = reader.ReadUInt16();
-                int textLength = reader.ReadUInt16();
+                int positionOffset = reader.Position;
 
-                if (textLength > 239)
+                textIds = stackalloc ushort[textCount];
```

Comment:
The counts can be big, and this would allow someone to make a malicious payload to try and crash the server.
Two things, first let's validate the counts coming in and make sure they are equal to what we expect.

Second, we need to allocate an array if the count is  too high, usually it's 1024 bytes. And lastly, we can GC uninitialize the arrays instead of just making them.